DriverDataTimelineUI/src/main/java/com/drivequant/drivekit/timeline/ui/component/roadcontext/RoadContextViewModel.kt

```diff
@@ -9,45 +9,45 @@ import com.drivequant.drivekit.common.ui.component.contextcard.DKContextCard
 import com.drivequant.drivekit.common.ui.component.contextcard.DKContextCardItem
 import com.drivequant.drivekit.common.ui.utils.DKDataFormatter
 import com.drivequant.drivekit.common.ui.utils.convertToString
-import com.drivequant.drivekit.databaseutils.entity.DKRawTimeline
+import com.drivequant.drivekit.databaseutils.entity.RoadContext
+import com.drivequant.drivekit.driverdata.timeline.DKDriverTimeline
 import com.drivequant.drivekit.timeline.ui.R
 import com.drivequant.drivekit.timeline.ui.component.roadcontext.enum.EmptyRoadContextType
 import com.drivequant.drivekit.timeline.ui.component.roadcontext.enum.TimelineRoadContext
-import com.drivequant.drivekit.timeline.ui.distanceByRoadContext
-import com.drivequant.drivekit.timeline.ui.hasData
-import com.drivequant.drivekit.timeline.ui.totalDistanceForAllContexts
+import com.drivequant.drivekit.timeline.ui.component.roadcontext.enum.toTimelineRoadContext
+import java.util.Date
 
 internal class RoadContextViewModel : ViewModel(), DKContextCard {
 
     val changeObserver: MutableLiveData<Any> = MutableLiveData()
 
-    private var distanceByContext = mapOf<TimelineRoadContext, Double>()
-        set(value) {
-            field = value.filterNot { it.value <= 0.0 }
-        }
-
+    private var distanceByContext = mutableMapOf<TimelineRoadContext, Double>()
     private var contextCards: List<DKContextCardItem> = emptyList()
     private var totalDistanceForAllContext = 0.0
-    private var distance = 0.0
     private var emptyRoadContextType: EmptyRoadContextType? = null
 
-    fun configure(timeline: DKRawTimeline?, selectedIndex: Int?) {
-        if (timeline != null && selectedIndex != null && timeline.hasData()) {
-            this.totalDistanceForAllContext = timeline.totalDistanceForAllContexts(selectedIndex)
-            this.distanceByContext = timeline.distanceByRoadContext(selectedIndex)
-            this.distance = 0.0
-            this.distanceByContext.forEach {
-                this.distance += it.value
+    fun configure(timeline: DKDriverTimeline?, selectedDate: Date?) {
+        if (timeline != null && selectedDate != null) {
+            this.distanceByContext = mutableMapOf()
+            this.totalDistanceForAllContext = 0.0
+
+            timeline.roadContexts.forEach { it ->
+                val distance = it.value.firstOrNull { it.date == selectedDate }?.distance ?: 0.0
+                if (distance > 0) {
+                    this.distanceByContext[it.key.toTimelineRoadContext()] = distance
+                }
             }
```

Comment:
Je trouve un peu moyen le fait de faire `this.distanceByContext = mutableMapOf()` puis d'ajouter les éléments.
Je pense qu'à la ligne 24 on peut remplacer par : `private var distanceByContext = mapOf<TimelineRoadContext, Double>()` et modifier ici comme ça :
```suggestion
            this.totalDistanceForAllContext = 0.0
            this.distanceByContext = buildMap {
                timeline.roadContexts.forEach { roadContextItem ->
                    val distance = roadContextItem.value.firstOrNull { it.date == selectedDate }?.distance ?: 0.0
                    if (distance > 0) {
                        this[roadContextItem.key.toTimelineRoadContext()] = distance
                    }
                }
            }
```

Reply:
d'ailleurs le `this.totalDistanceForAllContext = 0.0` est inutile

Reply:
Je ne connaissais pas `buildMap`, merci

Reply:
De rien ;)